frontend/src/components/Accordion.tsx

```diff
@@ -0,0 +1,50 @@
+import { FC, ReactNode, useEffect, useRef, useState } from "react";
+
+import selectFAQ from "../assets/selectFAQ.svg";
+import unselectFAQ from "../assets/unselectFAQ.svg";
+import styles from "../stylesheets/Home.module.css";
+
+export type AccordionProps = {
+  title: string;
+  children?: ReactNode;
+  toggleAll: boolean;
+};
+
+export const Accordion: FC<AccordionProps> = ({ title, children, toggleAll }) => {
+  const [toggle, setIsOpen] = useState(toggleAll);
+  const [height, setHeight] = useState(toggleAll || toggle ? "auto" : "0px");
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (contentRef.current) {
+      setHeight(toggle ? `${contentRef.current.scrollHeight}px` : "0px");
+    }
+  }, [toggle, toggleAll]);
```

Comment:
Why does this component need to keep track of height? Not sure if it is necessary considering the DOM can usually handle the resizing when showing and hiding elements automatically.